monkey/monkey_island/cc/repositories/plugin_archive_parser.py

```diff
@@ -26,27 +35,59 @@ def tarinfo_type(tar: TarInfo) -> str:
     return "unknown"
 
 
-def parse_plugin(file: BinaryIO) -> AgentPlugin:
+def parse_plugin(file: BinaryIO, data_dir: Path) -> Mapping[OperatingSystem, AgentPlugin]:
     """
     Load a plugin from a tar file.
 
     :raises ValueError: If the file is not a valid plugin
     """
+
+    parsed_plugin = {}
+
     tar_file = TarFile(fileobj=file)
     try:
         manifest = get_plugin_manifest(tar_file)
         schema = get_plugin_schema(tar_file)
         source = get_plugin_source(tar_file)
+
+        extracted_plugin = _safe_extract_file(tar=tar_file, filename="plugin.tar")
+        plugin_source_tar = tarfile.TarFile(fileobj=io.BytesIO(extracted_plugin.read()))
+        plugin_source_vendors = get_plugin_source_vendors(plugin_source_tar)
+
+        if "vendor" in [vendor.name for vendor in plugin_source_vendors]:
+            # if vendor/ exists, we don't want to check if vendor-linux/ or vendor-windows/
+            # exist
+            plugin = AgentPlugin(
+                plugin_manifest=manifest,
+                config_schema=schema,
+                source_archive=source,
+                host_operating_systems=(OperatingSystem.LINUX, OperatingSystem.WINDOWS),
+            )
+            parsed_plugin[OperatingSystem.LINUX] = plugin
+            parsed_plugin[OperatingSystem.WINDOWS] = plugin
+
+            return parsed_plugin
+
+        # create dir to store files and plugins per OS
+        parsed_plugins_dir = data_dir / "parsed_plugins"
+        create_secure_directory(parsed_plugins_dir)
+
+        os_specific_plugin_source_archives = get_os_specific_plugin_source_archives(
+            plugin_source_tar, plugin_source_vendors, parsed_plugins_dir
+        )
+        for os_, os_specific_plugin_source_archive in os_specific_plugin_source_archives.items():
+            parsed_plugin[os_] = AgentPlugin(
+                plugin_manifest=manifest,
+                config_schema=schema,
+                source_archive=os_specific_plugin_source_archive,
+                host_operating_systems=(os_,),
+            )
+
+        return parsed_plugin
```

Comment:
This should be a method called `parse_plugin_with_different_vendors` or something like that